billing/views.py

```diff
@@ -185,11 +187,44 @@ def create_account_complete(request):
 
 @login_required
 def profile_billing_accounts(request):
+    billing_accounts = request.user.owned_billing_accounts.all()
+    update_success = False
+    form = UpdatePurchaseOrderForm()
+    ba_id = None
+    if request.method == "POST":
+        form = UpdatePurchaseOrderForm(request.POST)
+        if form.is_valid():
+            ba_id = form.cleaned_data["ba_id"]
+            updated_billing_account = get_object_or_404(BillingAccount, id=ba_id)
+            if updated_billing_account in billing_accounts:
```

Comment:
Can you add a comment here to say that this check is to make sure the user has permission to change the billing account? It wasn't immediately obvious to me, and so someone looking at it in future might not realise that this line is what protects this form against an authorization bypass.